sale_product_set_variants/tests/test_product_set.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+from odoo import exceptions
+from odoo.tests import common
+
+
+class test_product_set(common.TransactionCase):
+    """ Test Product set"""
+
+    def setUp(self):
+        super(test_product_set, self).setUp()
+        self.sale_order = self.env['sale.order']
+        self.product_set_add = self.env['product.set.add']
+
+    def _get_wizard(self, so, product_set):
+        so_set = self.product_set_add.with_context(
+            active_id=so.id).create({'product_set_id': product_set.id, })
+        so_set._onchange_product_set_id()
+        return so_set
+
+    def test_add_set(self):
+        so = self.env.ref('sale.sale_order_6')
+        count_lines = len(so.order_line)
+
+        product_set = self.env.ref(
+            'sale_product_set_variants.product_set_i5_computer')
+        # Simulation the opening of the wizard and adding a set on the
+        # current sale order
+        so_set = self._get_wizard(so, product_set)
+        so_set.add_set()
+        # checking our sale order
+        self.assertEquals(len(so.order_line), count_lines + 2)
+        self.assertEquals(so.amount_untaxed, 2300.4)
```

Comment:
better to use `assertAlmostEqual` in such cases

Reply:
Could not this 2300.4 value be calculated ?
Instead of landing like a magical one.